crates/worktree/src/worktree_tests.rs

```diff
@@ -1984,6 +1984,58 @@ fn test_unrelativize() {
     );
 }
 
+#[gpui::test]
+async fn test_gitignore_with_git_info_exclude(cx: &mut TestAppContext) {
+    init_test(cx);
+
+    // Test that .git/info/exclude entries are properly recognized
+    let fs = FakeFs::new(cx.background_executor.clone());
+    fs.insert_tree(
```

Comment:
Once the logic to find the global gitignore path is encapsulated by `Fs`, it would be great to add a global gitignore here too!